### ISSUE TITLE
Teletext OSD size

### ISSUE THREAD

**Issue by ua0lnj** [vdr-projects/vdr-plugin-osdteletext#7]
I would suggest, instead of explicitly setting the length and height of the OSD in the plugin setup, use OsdWidth() and OsdHeight().
And also change OSDHAlign & OSDVAlign to something else, need to think about what.
For example, OSDHAlign & OSDVAlign symmetrically reduce OSD size on both sides, not just offset on one side.

**Comment by pbiering** [comment]
- Potentially something like % would be better instead of pixel sizing?
- Also dropping HAlign and VAlign and always use center based % sizing?
do you like to contribute a PR?

**Comment by ua0lnj** [comment]
While only thoughts, we'll see.

**Comment by pbiering** [comment]
at least user gets now info via OSD if plugin has an OSD area issue: https://github.com/vdr-projects/vdr-plugin-osdteletext/pull/9

**Comment by pbiering** [comment]
imho options should be changed to
- vertical % of fullscreen (default 100)
- horizontal aspect
  - full = OSDWidth() * vertical%
  - ratiowise 720x576 mapped to 540x576 (for 5:4 OSD horizontal stretched on FullHD) to get back 4:3/5:4 box for Teletext

**Comment by pbiering** [comment]
first implementation done: https://github.com/pbiering/vdr-plugin-osdteletext/tree/rework-OSD-size
replaced halign/valign/width/height by h+v percent

working well, have only to investigate black bar on right/bottom, will update next

**Comment by pbiering** [comment]
branch updated, made OSD % calculation optional - please review

**Comment by pbiering** [comment]
hint would be great how to dynamically hide menu entries depending on the "OSDsizePctMode" toggle

**Comment by pbiering** [comment]
toggle implemented, but it become only active once menu is closed and reopened -> TODO would be get it instantly in action

**Comment by ua0lnj** [comment]
[display.c.diff.gz](https://github.com/vdr-projects/vdr-plugin-osdteletext/files/6022265/display.c.diff.gz)
Fixed black bar on right/bottom.
Fixed fallback to bpp=8, IsTrueColor() not works while  osd->SetAreas() not use.

I would suggest removing old halign/valign/width/height settings.

**Comment by pbiering** [comment]
Fallback for bpp=8 applied

The black bars disappeared after I've added special alignment code above "// align with hChars/vLines in case of less than 100 %"

Also would this change not potentially result in overrun, in case of Pct==100 in case of Left/Top > 0

```
-        x0 = (cOsd::OsdWidth() - OSDwidth) / 2;
-        y0 = (cOsd::OsdHeight() - OSDheight) / 2;
+        x0 = cOsd::OsdLeft() + (cOsd::OsdWidth() - OSDwidth) / 2;
+        y0 = cOsd::OsdTop() + (cOsd::OsdHeight() - OSDheight) / 2;
```

legacy settings can be removed also next

**Comment by pbiering** [comment]
further fix of your patch was required, now it's working well: https://github.com/pbiering/vdr-plugin-osdteletext/commit/c4400c900b5a26f8dc1d720613bf2ed18822d492

**Comment by ua0lnj** [comment]
I didn't put it that way, not black bar.

```
-        x0 = (cOsd::OsdWidth() - OSDwidth) / 2;
-        y0 = (cOsd::OsdHeight() - OSDheight) / 2;
+        x0 = cOsd::OsdLeft() + (cOsd::OsdWidth() - OSDwidth) / 2;
+        y0 = cOsd::OsdTop() + (cOsd::OsdHeight() - OSDheight) / 2;
```

This is need to center the picture, do 50% in height and 50% in width and you will see the difference.

**Comment by pbiering** [comment]
understood, did not had this test case configured (OSD was on 100% size)...fixed, please check latest version of branch, legacy configuration was now also removed.

**Comment by ua0lnj** [comment]
Now all fine.
Tested with softhddevice, softhdcuvid, xineliboutput, reelbox+eHD.
1920x1080 and 720x756.

**Comment by pbiering** [comment]
thank you for testing, PR now included into master, new release will be created next: 1.0.0

**Comment by pbiering** [comment]
just note that some issues were found and complains were received: https://www.vdr-portal.de/forum/index.php?thread/134254-osdteletext-1-0-0-released/&postID=1336776#post1336787

Will be fixed by:
- https://github.com/pbiering/vdr-plugin-osdteletext/tree/re-add-h-v-osd-shift

And also a new feature was developed:
- https://github.com/pbiering/vdr-plugin-osdteletext/tree/24-line-mode

Feel free to test
